plugins/out_stackdriver/stackdriver.c

```diff
@@ -2637,6 +2728,10 @@ static void cb_stackdriver_flush(struct flb_event_chunk *event_chunk,
         }
         else if (c->resp.status >= 400 && c->resp.status < 500) {
             ret_code = FLB_ERROR;
+            if(c->resp.status == 400){
+              ret = parse_partial_success_response(c, ctx, &partial_failure);
+              cmt_counter_add(ctx->cmt_failed_requests, ts, (double)partial_failure, 1, (char *[]) {name});
```

Comment:
We're still counting this as one "failed request". We're already going to be increasing this counter further down this function. 

The number of failures in the partial failure should be added to [cmt_dropped_records](https://github.com/fluent/fluent-bit/blob/3f93f98a2902cb9f0fcf354a8c720e6ae4ee2559/include/fluent-bit/flb_output.h#L361). You can access that counter through `ctx->ins->cmt_dropped_records`.